test/functional/test-intersection-observer-polyfill.js

```diff
@@ -0,0 +1,461 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IntersectionObserverPolyfill,
+} from '../../src/intersection-observer-polyfill';
+import {layoutRectLtwh} from '../../src/layout-rect';
+import * as sinon from 'sinon';
+
+describe('IntersectionObserverPolyfill', () => {
+  let sandbox;
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('getThresholdSlot_', () => {
```

Comment:
this is implementation detail. i don't think we need a test here.
rather, we should test if intersection is fired at the right tick.